File: Data Structures & Algorithms/binary-search/submission-7.java
class Solution {
    public int search(int[] nums, int target) {
        int l=0,r=nums.length-1;
        if(nums.length==1 && nums[0]==target)
        {
            return 0;
        }
        while(l<=r)
        {
            int gen=(l+r)/2;
            if(nums[gen]<target)
            {
                l++;
            }
            else if(nums[gen]>target)
            {
                r--;
            }
           
            else{
                return gen;
            }
        }
        return -1;
    }
}

File: Data Structures & Algorithms/two-integer-sum/submission-7.py
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        freq={}
        for n in range(0,len(nums)):
            diff = target-nums[n]
            if diff in freq:
                if freq[diff]<n: return [freq[diff],n]
                return [n,freq[diff]]
            freq[nums[n]]=n
        return []
